core/fetch_user_info.go

```diff
@@ -1,9 +1,19 @@
 package core
 
-import "net/http"
+import (
+	"fmt"
+	"net/http"
+)
 
 func FetchUserInfo(client *http.Client, userInfoEndpoint, accessToken string) (UserInfoResponse, error) {
-	response, requestErr := client.Get(userInfoEndpoint)
+	req, err := http.NewRequest("GET", userInfoEndpoint, nil)
```

Comment:
Prefer using a specific err variable rather than overwriting it in the later code.
```suggestion
	req, createRequestErr := http.NewRequest("GET", userInfoEndpoint, nil)
```